src/hd/mod_hd_phys.t

```diff
@@ -652,70 +652,90 @@ contains
     use mod_global_parameters
     use mod_viscosity, only: visc_add_source_geom ! viscInDiv
-
+    use mod_dust, only: dust_n_species, dust_mom, dust_rho
     integer, intent(in)             :: ixI^L, ixO^L
     double precision, intent(in)    :: qdt, x(ixI^S, 1:ndim)
     double precision, intent(inout) :: wCT(ixI^S, 1:nw), w(ixI^S, 1:nw)
     ! to change and to set as a parameter in the parfile once the possibility to
     ! solve the equations in an angular momentum conserving form has been
     ! implemented (change tvdlf.t eg)
-    double precision :: tmp(ixI^S),tmp1(ixI^S)
+    double precision :: pth(ixI^S), source(ixI^S)
     integer                         :: iw,idir, h1x^L{^NOONED, h2x^L}
     integer :: mr_,mphi_ ! Polar var. names
+    integer :: irho, ifluid, n_fluids = 1
 
-    mr_=mom(1); mphi_=mom(1)-1+phi_ ! Polar var. names
+    if (hd_dust) n_fluids = 1 + dust_n_species
 
     select case (typeaxial)
     case ("cylindrical")
-       ! s[mr]=(pthermal+mphi**2/rho)/radius
-       call hd_get_pthermal(wCT,x,ixI^L,ixO^L,tmp)
-       if(phi_>0) then
-         tmp(ixO^S)=tmp(ixO^S)+wCT(ixO^S,mphi_)**2/wCT(ixO^S,rho_)
-         w(ixO^S,mr_)=w(ixO^S,mr_)+qdt*tmp(ixO^S)/x(ixO^S,1)
-         ! s[mphi]=(-mphi*mr/rho)/radius
-         ! Ileyk : beware the index permutation : mphi=2 if -phi=2 (2.5D
-         ! (r,theta) grids) BUT mphi=3 if -phi=3 (for 2.5D (r,z) grids)
-         if(.not. angmomfix) then
-           tmp(ixO^S)=-wCT(ixO^S,mphi_)*wCT(ixO^S,mr_)/wCT(ixO^S,rho_)
-           w(ixO^S,mphi_)=w(ixO^S,mphi_)+qdt*tmp(ixO^S)/x(ixO^S,1)
-         end if
-       else
-         ! s[mr]=2pthermal/radius
-         w(ixO^S,mr_)=w(ixO^S,mr_)+qdt*tmp(ixO^S)/x(ixO^S,1)
-       end if
+       do ifluid = 0, n_fluids-1
+          ! s[mr]=(pthermal+mphi**2/rho)/radius
+          if (ifluid == 0) then
+             ! gas
+             irho  = rho_
+             mr_   = mom(r_)
+             mphi_ = mom(phi_)
+             call hd_get_pthermal(wCT, x, ixI^L, ixO^L, source)
+          else
+             ! dust : no pressure
+             irho  = dust_rho(ifluid)
+             mr_   = dust_mom(ifluid, r_)
+             mphi_ = dust_mom(ifluid, phi_)
+             source(ixI^S) = zero
+          end if
+          if (phi_ > 0) then
+             source(ixO^S) = source(ixO^S) + wCT(ixO^S, mphi_)**2 / wCT(ixO^S, irho)
+             w(ixO^S, mr_) = w(ixO^S, mr_) + qdt * source(ixO^S) / x(ixO^S, r_)
+             ! s[mphi]=(-mphi*mr/rho)/radius
+             if(.not. angmomfix) then
+                source(ixO^S) = -wCT(ixO^S, mphi_) * wCT(ixO^S, mr_) / wCT(ixO^S, irho)
+                w(ixO^S, mphi_) = w(ixO^S, mphi_) + qdt * source(ixO^S) / x(ixO^S, r_)
+             end if
+          else
+             ! s[mr]=2pthermal/radius
+             w(ixO^S, mr_) = w(ixO^S, mr_) + qdt * source(ixO^S) / x(ixO^S, r_)
+          end if
+       end do
     case ("spherical")
+       ! Clement : case (hd_dust == .true.) not implemented yet
```

Comment:
Feel free to add a stop statement (e.g. call mpistop("... not implemented")) for this case

Reply:
Right. It would have saved me so much time if this had been present from the start, I should have thought of putting it in myself.